File: palindrome_number/palindrome_number.c
#include <stdbool.h>

bool isPalindrome(int x){
	if(x < 0 || (x != 0 && x % 10 == 0)) return false;
	
	long int rev = 0;
	while(x > rev) {
		rev = rev * 10 + x % 10;
		x /= 10;
	}
			
	return (x == rev || x == rev / 10);
}

File: palindrome_number/palindrome_number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
	x = String(x);
	var h = x.length / 2;
	for (var i = 0; i < h; i++) {
		if (x[i] !== x[x.length - i - 1]) return false;
	}
	return true;
};
